node/pkg/processor/benchmark_test.go

```diff
@@ -0,0 +1,243 @@
+package processor
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"fmt"
+	"os"
+	"runtime/pprof"
+	"testing"
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/common"
+	"github.com/certusone/wormhole/node/pkg/db"
+	"github.com/certusone/wormhole/node/pkg/gwrelayer"
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+
+	// gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+	ethCommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+	"go.uber.org/zap"
+)
+
+/*
+No gwrelayer:
+average time to do  1800000  observations:  51.246µs
+there were  1100000  under quorum, taking an average time of  62.888µs
+there were  100000  quorum reached, taking an average time of  68.049µs
+there were  600000  over quorum, taking an average time of  27.101µs
+there were  100000  handle message calls, taking an average time of  28.723µs
+
+With gwrelayer (where observations are not relayed):
+average time to do  1800000  observations:  51.18µs
+there were  1100000  under quorum, taking an average time of  62.713µs
+there were  100000  quorum reached, taking an average time of  68.316µs
+there were  600000  over quorum, taking an average time of  27.182µs
+there were  100000  handle message calls, taking an average time of  28.704µs
+*/
+
+// go test -bench ^BenchmarkHandleObservation -benchtime=1x
+func BenchmarkHandleObservation(b *testing.B) {
+	const NumObservations = 100000
+	ctx := context.Background()
+	db := db.OpenDb(nil, nil)
+	defer db.Close()
+	p, pd := createProcessorForTest(b, NumObservations, ctx, db, false)
+	require.NotNil(b, p)
+	require.NotNil(b, pd)
+
+	var totalTime, underQuorumTime, quorumReachedTime, overQuorumTime, handleMsgTime time.Duration
+	var totalCount, underQuorumCount, quorumReachedCount, overQuorumCount int
+	for count := 0; count < NumObservations; count++ {
+		k := pd.createMessagePublication(b, uint64(count))
+		start := time.Now()
+		p.handleMessage(k)
+		handleMsgTime += time.Since(start)
+
+		for guardianIdx := 1; guardianIdx < 19; guardianIdx++ {
```

Comment:
since `guardianIdx` starts from `1`, should the terminating condition be `guardianIdx <= 19`?

Reply:
No, because "we" are guardian zero. Guardians 1-18 are the ones from whom we receive observations.